aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2147,6 +2147,53 @@ Tensor& take_along_dim_out(const Tensor& self, const Tensor& indices, std::optio
   return at::gather_out(result, self.view(-1), 0, indices.view(-1));
 }
 
+namespace {
+
+inline std::tuple<Tensor, Tensor> _put_along_dim_helper(const Tensor& self, const Tensor& indices, const Tensor& values, int64_t dim) {
+  auto new_indices_sizes = DimVector(indices.sizes());
+  for (int i = 0; i < indices.dim(); i++) {
+    if (i == dim) {
+      new_indices_sizes[i] = indices.size(i);
+    } else {
+      new_indices_sizes[i] = self.size(i);
+    }
+  }
```

Comment:
```suggestion
  auto new_indices_sizes = DimVector(self.sizes());
  new_indices_sizes[dim] = indices.size(i);
```

Reply:
updated it in [b5a43f5](https://github.com/pytorch/pytorch/pull/125601/commits/b5a43f59ce6f1082bdc3868949be79c337f159bd)